{
  Map<String,Exception> failures=null;
  Collection<String> cores=null;
  Exception fail=null;
  init("empty_flow");
  File solrXml=new File(solrHome,"solr.xml");
  FileUtils.write(solrXml,EMPTY_SOLR_XML,IOUtils.CHARSET_UTF_8.toString());
  cc.load(solrHome.getAbsolutePath(),solrXml);
  cores=cc.getCoreNames();
  assertNotNull("core names is null",cores);
  assertEquals("wrong number of cores",0,cores.size());
  failures=cc.getCoreInitFailures();
  assertNotNull("core failures is a null map",failures);
  assertEquals("wrong number of core failures",0,failures.size());
  final CoreDescriptor bogus=new CoreDescriptor(cc,"bogus","bogus_path");
  try {
    ignoreException(Pattern.quote("bogus_path"));
    cc.create(bogus);
    fail("bogus inst dir failed to trigger exception from create");
  }
 catch (  SolrException e) {
    assertTrue("init exception doesn't mention bogus dir: " + e.getCause().getCause().getMessage(),0 < e.getCause().getCause().getMessage().indexOf("bogus_path"));
  }
  cores=cc.getCoreNames();
  assertNotNull("core names is null",cores);
  assertEquals("wrong number of cores",0,cores.size());
  failures=cc.getCoreInitFailures();
  assertNotNull("core failures is a null map",failures);
  assertEquals("wrong number of core failures",1,failures.size());
  fail=failures.get("bogus");
  assertNotNull("null failure for test core",fail);
  assertTrue("init failure doesn't mention problem: " + fail.getCause().getMessage(),0 < fail.getCause().getMessage().indexOf("bogus_path"));
}
